src/error.rs

```diff
@@ -271,45 +287,19 @@ pub enum ErrorKind {
     #[error("Command failed {0}")]
     CommandError(CommandError),
 
-    // `trust_dns` does not implement the `Error` trait on their errors, so we have to manually
-    // implement `From` rather than using the `source annotation.
-    /// Wrapper around `trust_dns_resolver::error::ResolveError`.
-    #[error("{0}")]
-    DnsResolve(trust_dns_resolver::error::ResolveError),
+    /// An error occurred during DNS resolution.
+    #[error("An error occurred during DNS resolution: {message}")]
+    #[non_exhaustive]
+    DnsResolveError { message: String },
 
     #[error("Internal error: {message}")]
     #[non_exhaustive]
     InternalError { message: String },
 
-    /// Wrapper around `webpki::InvalidDNSNameError`.
-    #[error("{0}")]
-    InvalidDnsName(#[from] webpki::InvalidDNSNameError),
-
-    /// A hostname could not be parsed.
-    #[error("Unable to parse hostname: {hostname}")]
-    #[non_exhaustive]
-    InvalidHostname { hostname: String },
-
     /// Wrapper around [`std::io::Error`](https://doc.rust-lang.org/std/io/struct.Error.html).
     #[error("{0}")]
     Io(Arc<std::io::Error>),
 
-    #[error("No DNS results for domain {0}")]
-    NoDnsResults(StreamAddress),
-
-    /// A database operation failed to send or receive a reply.
-    #[error("A database operation failed to send or receive a reply: {message}")]
-    #[non_exhaustive]
-    OperationError { message: String },
```

Comment:
All uses of this were rewritten as either `ResponseError` or `InternalError`, so this wasn't needed anymore.